node/node_handler.go

```diff
@@ -47,11 +52,15 @@ func (node *Node) processSkippedMsgTypeByteValue(
 }
 
 var (
-	errInvalidPayloadSize         = errors.New("invalid payload size")
-	errWrongBlockMsgSize          = errors.New("invalid block message size")
-	latestSentCrosslinkNum uint64 = 0
+	errInvalidPayloadSize = errors.New("invalid payload size")
+	errWrongBlockMsgSize  = errors.New("invalid block message size")
+	latestSentCrosslink   *utils.CrosslinkStorage
```

Comment:
Why change to a storage with 4 slots? Here we only need to keep track of the latest sent cross link for the validator's own shard. So one number is enough.